packages/core/src/services/evaluationResults/computeEvaluationResultsByDocumentContent.ts

```diff
@@ -98,98 +75,34 @@ export async function computeEvaluationResultsByDocumentContent(
     db,
   )
 
-  const limitedEvaluationResults = db
+  const rows = await db
     .select({
       id: evaluationResultsScope.id,
-      documentLogId: evaluationResultsScope.documentLogId,
       source: evaluationResultsScope.source,
       result: evaluationResultsScope.result,
       createdAt: evaluationResultsScope.createdAt,
     })
     .from(evaluationResultsScope)
-    .where(eq(evaluationResultsScope.evaluationId, evaluation.id))
-    .orderBy(
-      desc(evaluationResultsScope.createdAt),
-      desc(evaluationResultsScope.id),
-    )
-    .limit(pageSize)
-    .offset(offset)
-    .as('limited_evaluation_results')
-
-  const rows = await db
-    .select({
-      id: limitedEvaluationResults.id,
-      source: limitedEvaluationResults.source,
-      result: limitedEvaluationResults.result,
-      createdAt: limitedEvaluationResults.createdAt,
-    })
-    .from(limitedEvaluationResults)
     .innerJoin(
       documentLogsScope,
-      eq(documentLogsScope.id, limitedEvaluationResults.documentLogId),
+      eq(documentLogsScope.id, evaluationResultsScope.documentLogId),
     )
     .innerJoin(commits, eq(commits.id, documentLogsScope.commitId))
     .where(
-      or(
-        eq(documentLogsScope.contentHash, hashContent(resolvedContent)),
-        eq(commits.id, commit.id),
+      and(
+        eq(evaluationResultsScope.evaluationId, evaluation.id),
+        or(
+          eq(documentLogsScope.contentHash, hashContent(resolvedContent)),
+          eq(commits.id, commit.id),
+        ),
       ),
     )
     .orderBy(
-      desc(limitedEvaluationResults.createdAt),
-      desc(limitedEvaluationResults.id),
+      desc(evaluationResultsScope.createdAt),
+      desc(evaluationResultsScope.id),
     )
+    .limit(pageSize)
+    .offset(offset)
 
   return Result.ok(rows)
 }
-
-export async function computeEvaluationResultsByDocumentContentCount(
```

Comment:
Should we also remove from logs?

Reply:
hmm not really if it's just a count of document logs filtered by documentuuid and commit it should be fast, what's hurting this particular count is the multiples inner joins required